### PR TITLE
Feature heart health

### PR DESCRIPTION
**Pull request by Basler182**
# *Heart Health*

## :gear: Release Notes 

**New Heart Health Data Page:**
Consists of 4 subpages:

- Symptoms
- Heart Rate
- Blood Pressure
- Weight

**Each subpage includes:**

- A chart for data visualization
- A table displaying detailed data
- Filtering options to refine data display

**New Repository:**

- A new repository has been created for data retrieval, ensuring efficient and seamless access to the necessary data for visualization.

## :white_check_mark: Testing
as soon as the basic structure has been checked, further tests are added

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by eldcn — COMMENTED** [review]
I am just publishing my pending comments from yesterday. We aligned with @pauljohanneskraft today in the lab, and he did some adaptions in several PRs including this [one](https://github.com/StanfordBDHG/ENGAGE-HF-Firebase/pull/57). I already pushed some adjustments in this PR, and will continue the other open points next.

FYI: You can use the seeded data with the firebase emulator from https://github.com/StanfordBDHG/ENGAGE-HF-Firebase. Clone the repository, navigate to `cd functions` and just run `npm run serve:seeded` to setup the emulator. I used `engagehf-patient0@stanford.edu` user.

**Review by eldcn — APPROVED** [review]

File: app/src/main/kotlin/edu/stanford/bdh/engagehf/health/HealthPage.kt [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+            is HealthUiState.Success -> {
+                if (uiState.data.records.isEmpty()) {
+                    Text(
+                        text = "No data available",
+                        style = TextStyles.headlineMedium
+                    )
+                } else {
+                    HealthHeader(
+                        uiState.data.headerData,
```
[the remark this answers is not in the document]
Comment (Basler182) [reply app/src/main/kotlin/edu/stanford/bdh/engagehf/health/HealthPage.kt]: good idea! :) it's a pity that you can't set it to do this automatically

File: app/src/main/kotlin/edu/stanford/bdh/engagehf/health/HealthPage.kt [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    onAction: (HealthAction) -> Unit,
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .verticalScroll(rememberScrollState())
+            .testIdentifier(HealthPageTestIdentifier.ROOT)
+    ) {
+        when (uiState) {
+            is HealthUiState.Error -> {
```
[the remark this answers is not in the document]
Comment (Basler182) [reply app/src/main/kotlin/edu/stanford/bdh/engagehf/health/HealthPage.kt]: good idea, this prevents the boilercode in the places and works great :)

File: app/src/main/kotlin/edu/stanford/bdh/engagehf/health/HealthPage.kt [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                    modifier = Modifier.testIdentifier(HealthPageTestIdentifier.ERROR_MESSAGE)
+                )
+                Spacer(
+                    modifier = Modifier
+                        .weight(1f)
+                        .fillMaxHeight()
+                )
+            }
+
+            HealthUiState.Loading -> {
```
[the remark this answers is not in the document]
Comment (Basler182) [reply app/src/main/kotlin/edu/stanford/bdh/engagehf/health/HealthPage.kt]: yes that sounds good

**Pull request merged**